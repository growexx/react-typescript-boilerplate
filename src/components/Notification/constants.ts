import { getNotificationsSuccessMock } from "./stub";

export const NOTIFICATION_LIMIT = 5;
export const getNotificationsMock = () => getNotificationsSuccessMock();
