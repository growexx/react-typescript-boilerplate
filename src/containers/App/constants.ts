import {
    FAV_ICON_152_152,
    FAV_ICON_16_16,
    FAV_ICON_32_32,
    APPLE_TOUCH_ICON,
    FAV_ICON_192_192,
} from '../../images/favicons';
export const LOAD_REPOS = 'boilerplate/App/LOAD_REPOS';
export const LOAD_REPOS_SUCCESS = 'boilerplate/App/LOAD_REPOS_SUCCESS';
export const LOAD_REPOS_ERROR = 'boilerplate/App/LOAD_REPOS_ERROR';
export const LOAD_APP = 'boilerplate/App/LOAD_APP';
export const FAV_ICONS = [
    {
        rel: 'icon',
        type: 'image/png',
        sizes: '512x512',
        href: FAV_ICON_152_152,
    },

    {
        rel: 'icon',
        type: 'image/png',
        sizes: '192x192',
        href: FAV_ICON_192_192,
    },
    {
        rel: 'icon',
        type: 'image/png',
        sizes: '32x32',
        href: FAV_ICON_32_32,
    },
    {
        rel: 'icon',
        type: 'image/png',
        sizes: '16x16',
        href: FAV_ICON_16_16,
    },
    {
        rel: 'apple-touch-icon',
        href: APPLE_TOUCH_ICON,
    },
];
