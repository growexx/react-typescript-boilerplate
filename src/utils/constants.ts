export const TOKEN_KEY = 'jwtToken';
export const USER_DATA_KEY = 'userData';
export const EMITTER_EVENTS = {
  LOG_IN: 'USER_LOGGED_IN',
  LOG_OUT: 'USER_LOGGED_OUT',
};
