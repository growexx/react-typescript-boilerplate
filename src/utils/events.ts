import EventEmitter from 'eventemitter3';

const eventEmitter = new EventEmitter();

type CallBackFunction = (payload: any) => void;

const Emitter = {
    on: (event: string, fn: CallBackFunction) => eventEmitter.on(event, fn),
    once: (event: string, fn: CallBackFunction) => eventEmitter.once(event, fn),
    off: (event: string, fn?: CallBackFunction) => eventEmitter.off(event, fn),
    emit: (event: string, payload?: any) => eventEmitter.emit(event, payload),
};

Object.freeze(Emitter);
export default Emitter;
