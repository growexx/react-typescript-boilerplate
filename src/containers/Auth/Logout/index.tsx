import React, { useEffect } from "react";

import { useNavigate } from "react-router";

import { EMITTER_EVENTS } from "../../../utils/constants";
import Emitter from "../../../utils/events";
import StorageService from "../../../utils/StorageService";
import { ROUTES } from "../../constants";

const Logout = () => {
  const navigate = useNavigate();

  useEffect(() => {
    StorageService.clear();
    Emitter.emit(EMITTER_EVENTS.LOG_OUT);
    navigate(ROUTES.LOGIN);

    return () => {
      Emitter.off(EMITTER_EVENTS.LOG_OUT);
    };
  }, [navigate]);

  return <React.Fragment />;
};

export default Logout;
