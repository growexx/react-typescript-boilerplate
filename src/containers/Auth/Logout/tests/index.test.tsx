import { render } from "@testing-library/react";
import { MemoryRouter as Router } from "react-router";

import Logout from "..";
import { EMITTER_EVENTS } from "../../../../utils/constants";
import Emitter from "../../../../utils/events";

const setup = () =>
  render(
    <Router>
      <Logout />
    </Router>
  );

describe("<Logout />", () => {
  test("Should render and match the snapshot", () => {
    const {
      container: { firstChild },
    } = setup();
    expect(firstChild).toMatchSnapshot();
  });

  test("Should emit logout event", () => {
    const logoutEvent = jest.fn();
    Emitter.on(EMITTER_EVENTS.LOG_OUT, logoutEvent);
    setup();
    expect(logoutEvent.mock.calls.length).toBe(1);
  });
});
