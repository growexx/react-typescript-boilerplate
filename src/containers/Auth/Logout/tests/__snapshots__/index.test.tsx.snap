// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Logout /> Should render and match the snapshot 1`] = `null`;
